src/backend/access/transam/xlog.c

```diff
@@ -10837,11 +10837,23 @@ xlog_redo(XLogReaderState *record)
 		/*
 		 * GPDB: Restore point records can act as a point of
 		 * synchronization to ensure cluster-wide consistency during WAL
-		 * replay. WAL replay is paused at each restore point until it is
+		 * replay. If the restore point is specified
+		 * in gp_pause_on_restore_point_replay guc
+		 * WAL replay is paused at the restore point until it is
 		 * explicitly resumed.
 		 */
-		if (gp_pause_on_restore_point_replay)
+		xl_restore_point *recordRestorePointData;
+		recordRestorePointData = (xl_restore_point *) XLogRecGetData(record);
+
+		if (strcmp(recordRestorePointData->rp_name, gp_pause_on_restore_point_replay) == 0)
+		{
+			ereport(LOG,
+					(errmsg("setting pause for restore point: '%s', created at time: %s",
+							recordRestorePointData->rp_name,
+							timestamptz_to_str(recordRestorePointData->rp_time))));
```

Comment:
Recommend changing the message:
```suggestion
					(errmsg("pausing on restore point '%s' created at on %s",
							recordRestorePointData->rp_name,
							timestamptz_to_str(recordRestorePointData->rp_time))));
```

But at the same time... why do we need to log out `rp_time`?  It doesn't seem too useful.

Reply:
Talking with Deep, we'll keep the message as "setting pause on restore point '%s'" and take out the rp_time part.